src/Panel/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -0,0 +1,496 @@
+
+  /**
+   *
+   * @param nextProps
+   * @param nextState
+   */
+  shouldComponentUpdate(nextProps: TimeLayerSliderPanelProps, nextState: TimeLayerSliderPanelState) {
+    const {
+      value,
+      autoPlayActive,
+      startDate,
+      endDate
+    } = this.state;
+    const {
+      timeAwareLayers
+    } = this.props;
+
+    if (nextState.value !== value) {
+      return true;
+    }
+    if (nextState.autoPlayActive !== autoPlayActive) {
+      return true;
+    }
+    if (nextState.startDate !== startDate) {
+      return true;
+    }
+    if (nextState.endDate !== endDate) {
+      return true;
+    }
+    if (!(_isEqual(nextProps.timeAwareLayers, timeAwareLayers))) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Wraps the TimeSlider component in timeLayerAware.
+   */
+  wrapTimeSlider = () => {
+    this._wmsTimeLayers = [];
+    this.props.timeAwareLayers!.forEach((l: any) => {
+      if (l.get('type') === 'WMSTime') {
+        this._wmsTimeLayers.push({
+          layer: l
+        });
+      }
+    });
+    // make sure an initial value is set
+    this.wmsTimeHandler(this.state.value);
+    this._TimeLayerAwareSlider = timeLayerAware(TimeSlider, this._wmsTimeLayers);
+  }
+
+  /**
+   * Updates slider time range depending on chosen layer set.
+   */
+  findRangeForLayers = () => {
+    const {
+      timeAwareLayers
+    } = this.props;
+    const {
+      startDate,
+      endDate,
+    } = this.state;
+
+    if (timeAwareLayers.length === 0) {
+      return;
+    }
+
+    let newStartDate: moment.Moment;
+    let newEndDate: moment.Moment;
+    let startDatesFromLayers: moment.Moment[] = [];
+    let endDatesFromLayers: moment.Moment[] = [];
+
+    this._wmsTimeLayers.forEach((l: any) => {
+      const layerStartDate = l.layer.get('startDate');
+      const layerEndDate = l.layer.get('endDate');
+      let sdm;
+      let edm;
+      if (layerStartDate) {
+        sdm = moment(l.layer.get('startDate'));
+      }
+      if (layerEndDate) {
+        edm = moment(l.layer.get('endDate'));
+      }
+      if (sdm) {
+        startDatesFromLayers.push(sdm);
+      }
+      if (edm) {
+        endDatesFromLayers.push(edm);
+      }
+    });
+    newStartDate = startDatesFromLayers.length > 0 ? moment.min(startDatesFromLayers) : startDate;
+    newEndDate = endDatesFromLayers.length > 0 ? moment.max(endDatesFromLayers) : endDate;
+    this.updateDataRange([newStartDate, newEndDate]);
+  }
+
+  /**
+   * Handler for the time slider change behaviour
+   */
+  timeSliderCustomHandler = (value: any) => {
+    const currentMoment = moment(value).milliseconds(0);
+    const newValue = currentMoment.clone();
+    this.setState({
+      value: newValue
+    });
+    if (this.props.onChange) {
+      this.props.onChange(newValue);
+    }
+  }
+
+  /**
+   * makes sure that the appended time parameter in GetMap calls
+   * is rounded to full hours to receive a valid response
+   */
+  wmsTimeHandler = (value?: any) => {
```

Comment:
Why do we need this? A WMS-Time may also accept a non-full-hours time parameter, see e.g. [here](https://docs.geoserver.org/stable/en/user/services/wms/time.html).

Reply:
I don't know, it was implemented that way in the react-geo-baseclient. It may have been needed to display images for a timespan of a complete hour, if there is only one image available for a solid hour. I have now added a query to the configuration of the layer, whether this behavior is desired or not (`roundToFullHours`).